core/src/main/java/tech/pegasys/web3signer/core/multikey/metadata/AzureSecretSigningMetadata.java

```diff
@@ -25,21 +30,27 @@
   private final String tenantId;
   private final String vaultName;
   private final String secretName;
+  private final AzureAuthenticationMode authenticationMode;
 
   @JsonCreator
   public AzureSecretSigningMetadata(
       @JsonProperty("clientId") final String clientId,
       @JsonProperty("clientSecret") final String clientSecret,
       @JsonProperty("tenantId") final String tenantId,
-      @JsonProperty("vaultName") final String vaultName,
-      @JsonProperty("secretName") final String secretName,
+      @JsonProperty(value = "vaultName", required = true) final String vaultName,
+      @JsonProperty(value = "secretName", required = true) final String secretName,
```

Comment:
they were always required, but we didn't set them to required earlier (so it would have failed with a NPE or similar exception upon constructing the signer). Now I added the required flag which are required by both auth modes. Rest of the options are custom validated .. for CLIENT_SECRET mode, we need all other options, for Managed_Identity, we optionally need clinetId only.